src/ig/IGExporter.ts

```diff
@@ -398,6 +399,29 @@ export class IGExporter {
     }
   }
 
+  addLinkReferences(igPath: string): void {
+    // no need to make this file if there are no resources
+    if (!this.ig.definition?.resource.length) {
+      return;
+    }
+    const linkReferencesDir = path.join(igPath, 'fsh-generated', 'includes');
+    const linkReferencesExportPath = path.join(linkReferencesDir, 'markdown-link-references.md');
+    ensureDirSync(linkReferencesDir);
+    const content = this.ig.definition.resource.map(igResource => {
+      // FSH resources and predefined resources will have a _linkRef
+      // a configured resource may lack a name
+      // in that case, try to build a useful name from the reference
+      const linkName =
+        igResource._linkRef ??
+        igResource.name ??
+        igResource.reference?.reference?.replace(/^[^\/]*\//, '');
+      // delete the _linkRef now that we've used it
+      delete igResource._linkRef;
+      return `[${linkName}]: ${igResource.reference?.reference?.replace('/', '-')}.html`;
```

Comment:
Are there cases there `ig.reference` or `ig.reference.reference` might be undefined? If so, might it make sense to skip those links rather than write them out as `undefined.html`?

Or do you think that outputting `undefined.html` might be better since it's a more visible failure than just supressing the link?

Or is it an impossible condition, so it doesn't matter?  I have a feeling it might be an impossible condition and there will always be a reference.  But I'm looking for confirmation I guess.

Reply:
I think it's an impossible condition, and I was using `?.` aggressively because these attributes are not guaranteed by the type definition. But based on the current implementation, I think that `igResource.reference.reference` is guaranteed to exist here.